service/pkg/config/environment_value_loader.go

```diff
@@ -51,17 +43,51 @@ func (l *EnvironmentValueLoader) Get(key string) (any, error) {
 			return nil, fmt.Errorf("environment value %s is not allowed", key)
 		}
 	}
-	return l.viper.Get(key), nil
+	value, found := l.loadedValues[key]
+	if found {
+		return value, nil
+	}
+	return nil, nil
 }
 
 // GetConfigKeys returns all the configuration keys found in the environment variables.
 func (l *EnvironmentValueLoader) GetConfigKeys() ([]string, error) {
-	return l.viper.AllKeys(), nil
+	return l.loadedKeys, nil
 }
 
 // Load loads the configuration into the provided struct
 func (l *EnvironmentValueLoader) Load(_ Config) error {
-	// For environment variables, Viper's `AutomaticEnv` handles this, so no explicit load is needed here.
+	var loadedKeys []string
+	loadedValues := make(map[string]string)
+
+	env := os.Environ()
+	for _, kv := range env {
+		upperKV := strings.ToUpper(kv)
+		if strings.HasPrefix(upperKV, l.envKeyPrefix) {
+			eqIdx := strings.Index(upperKV, "=")
```

Comment:
@ntrevino-virtru  should we check for `-1`  here if somehow the equals is missing